module/VuFind/src/VuFind/ILS/OAuth2Service.php

```diff
@@ -0,0 +1,122 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Class OAuth2Service
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Moravian Library 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  VuFind\ILS
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:ils_drivers Wiki
+ */
+namespace VuFind\ILS;
+
+use Laminas\Http\Client\Exception\RuntimeException as HttpException;
```

Comment:
Consider adding a new exception class in the \VuFind\Exception namespace.

Reply:
I added AuthToken exception class